### PR TITLE
Fitted FFD generation

### PR DESCRIPTION
**Pull request by sseraj**
## Purpose

This PR adds a function for generating fitted wing FFDs. The function uses DVCon to perform surface projections, which was @anilyil's idea. @ArshSaja made the initial implementation.

Here is an example of a C172 wing FFD

c172_fitted

I added the function to a file named ffd_generation.py and also moved `write_wing_FFD_file` to this file because I thought it was more appropriate than file_io.py.

## Expected time until merged

A week or possibly more if reviewers want to test it out on their own geometries

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Comment by ArshSaja** [comment]
Nice work @sseraj . Thank you for adding it

File: pygeo/geo_utils/ffd_generation.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    Examples
+    --------
+    >>> CFDSolver = ADFLOW(options=aeroOptions)
+    >>> surf = CFDSolver.getTriangulatedMeshSurface()
+    >>> surfFormat = "point-vector"
+    >>> outFile = "wing_ffd.xyz"
+    >>> nSpan = [4, 4]
+    >>> nChord = 8
+    >>> relMargins = [0.01, 0.001, 0.01]
+    >>> absMargins = [0.05, 0.001, 0.05]
```
Comment (marcomangano) [inline pygeo/geo_utils/ffd_generation.py:214]: I am a bit confused about the usage of relative and absolute margins. Are we taking the most conservative option? A one-line explanation in the docstrings above would help.

**Review by marcomangano — COMMENTED** [review]
Neat implementation! I just have one comment:

**Review by marcomangano — APPROVED** [review]

**Comment by joanibal** [comment]
How fitted are the FFDs? Can you post a picture of the embedding volume and with the points inside?

**Comment by ArshSaja** [comment]
This is the previous example I had.
exampleFFD

**Review by ArshSaja — APPROVED** [review]

**Comment by sseraj** [comment]
> How fitted are the FFDs? Can you post a picture of the embedding volume and with the points inside?

The fit will depend on the user-specified margins, so some manual tuning is required. For the C172 FFD, here is the embedded volume (blue) and the surface points.
embed1
embed2
embed3

**Pull request merged**